gerousiabot/discord_bot.py

```diff
@@ -0,0 +1,92 @@
+from typing import List
+
+from discord import Intents, Member, VoiceChannel, Client
+
+from gerousiabot import utils
+
+logger = utils.setup_logger()
+
+
+class GerousiaBot(Client):
+
+    def __init__(self, bot_token: str, **options):
+        intents = get_intents_needed_to_check_online_members()
+        logger.debug('Creating bot with with intents {}'.format(intents))
+
+        super().__init__(intents=intents, **options)
+        logger.debug('Creating bot with with valid token {}'.format(bot_token is not None))
+        self.bot_token = bot_token
+
+    async def on_ready(self):
+        """
+        Asynchronous event that fires once the bot has logged in and ready to work.
+        """
+        logger.info('Bot has logged in as {0.user}'.format(self))
+
+        server_id = int(utils.get_env_variable('GOOD_LOOKING_GAMERS_SERVER_ID'))
+        guild_voice_channels = await self.get_server_voice_channels(server_id)
+        members_in_voice_channels = await self.get_members_who_are_in_voice_channels(guild_voice_channels)
+
+        for member in members_in_voice_channels:
+            print(member)
+
+    async def get_members_who_are_in_voice_channels(self, voice_channels: List[VoiceChannel]) -> List[str]:
+        """
+        Gets all of the members who are in voice channels for a given server
+        :param voice_channels: a list of voice channels
+        :return: A list of strings with each members display name in Discord
+        """
+        voice_channels_with_members = list(
+            (filter(lambda voice_channel: len(voice_channel.members) != 0, voice_channels))
+        )
+
+        # This will be a list of lists
+        # [ [member_0, member_1], [member_2], etc ]
+        online_members = list(
+            map(lambda voice_channel: voice_channel.members, voice_channels_with_members)
+        )
+
+        # Converting from List of lists to a single List that has the members
+        flattened_online_members = [online_member for members in online_members for online_member in members]
```

Comment:
I usually use sum for this conversion - would it be useful here? flattened_online_members = sum(online_members, [])

Reply:
yeah that works. 
stealing that

Reply:
nice @KSBolton.